### PR TITLE
fix listener on follower if region changed

### PR DESCRIPTION
**Pull request by YuJuncen**
This is the commit `e0383539d` extracted from #25.

This fixed a bug that caused when region conf version changed, the follower may mistakenly listen on changes, also fixed the bugs in `SegmentMap` find at #22.

### CODE REVIEW

**Review by 3pointer — APPROVED** [review]
LGTM

File: components/br-stream/src/observer.rs [hunk 1]
```diff
@@ -229,6 +252,16 @@ mod tests {
...
 
+    macro_rules! assert_let {
+        (let $p:pat = $e:expr; $cc:tt) => {
+            if let $p = $e {
+                $cc
+            } else {
+                panic!("{} doesn't matches {}", stringify!($e), stringify!($p))
+            }
+        };
+    }
```
Comment (kennytm) [inline components/br-stream/src/observer.rs]: use `assert_matches`.

**Review by joccau — APPROVED** [review]
LGTM

File: components/br-stream/Cargo.toml [hunk 2]
```diff
@@ -52,5 +52,6 @@ online_config = { path = "../online_config" }
 resolved_ts = { path = "../resolved_ts"}
 
 [dev-dependencies]
+assert_matches = "1.5"
```
Comment (kennytm) [inline components/br-stream/Cargo.toml:55]: @YuJuncen i mean the standard `assert_matches` 😅 

https://doc.rust-lang.org/std/assert_matches/macro.assert_matches.html

**Pull request merged**